frontend/src/templates/Field/Table/ColumnCell.tsx

```diff
@@ -29,23 +29,29 @@ import { TableFieldInputs } from '../types'
 export interface ColumnCellProps
   extends UseTableCellProps<TableFieldInputs, string> {
   schemaId: string
+  responseMode: FormResponseMode
   columnSchema: ColumnDto
   colorTheme: FormColorTheme
 }
 
 export interface FieldColumnCellProps<T extends Column = Column> {
   schema: ColumnDto<T>
+  responseMode: FormResponseMode
   /** Represents `{schemaId}.{rowIndex}.{columnId}` */
   inputName: `${string}.${number}.${string}`
   colorTheme: FormColorTheme
 }
 
 const ShortTextColumnCell = ({
   schema,
+  responseMode,
   inputName,
   colorTheme,
 }: FieldColumnCellProps<ShortTextColumnBase>) => {
-  const rules = useMemo(() => createBaseValidationRules(schema), [schema])
+  const rules = useMemo(
+    () => createBaseValidationRules(schema, responseMode),
```

Comment:
We can't pass in `responseMode` here, `createBaseValidationRules` accepts a schema and a `disableRequiredValidation` boolean val

Reply:
Good catch, this is a change that's missed from [this comment](https://github.com/opengovsg/FormSG/pull/6960#discussion_r1426915229)

Reply:
cafd0762